### PR TITLE
In Sync(Couch|SQL)To(SQL|Couch), sync deletes

### PR DESCRIPTION
**Pull request by orangejenny**
@gcapalbo 

code buddy @benrudolph

### CODE REVIEW

**Comment by benrudolph** [comment]
cool, looks good to me. maybe a test for deletion?

**Comment by benrudolph** [comment]
(feel free to self merge, my comment is non-blocking)

File: dimagi/utils/couch/migration.py [hunk 1]
```diff
@@ -61,16 +61,23 @@ def _migration_automatically_handle_dups(self):
         """
         return False
 
-    def _migration_get_or_create_sql_object(self):
+    def _migration_get_sql_object(self):
         cls = self._migration_get_sql_model_class()
         try:
             obj = cls.objects.get(couch_id=self._id)
```
Comment (czue) [inline dimagi/utils/couch/migration.py]: small, but I think you could replace this with `return` and get rid of `obj`

**Comment by czue** [comment]
agree tests would be nice but agree no blockers

**Comment by orangejenny** [comment]
There are tests in https://github.com/dimagi/commcare-hq/pull/7579 - not ideal, since those tests will eventually be deprecated, but this code doesn't lend itself terribly well to more generic tests.

File: dimagi/utils/couch/migration.py [hunk 2]
```diff
@@ -120,16 +134,23 @@ def _migration_get_couch_model_class(cls):
         """
         raise NotImplementedError()
 
-    def _migration_get_or_create_couch_object(self):
+    def _migration_get_couch_object(self):
         cls = self._migration_get_couch_model_class()
         if not self.couch_id:
+            return None
+        else:
+            obj = cls.get(self.couch_id)
```
Comment (gcapalbo) [inline dimagi/utils/couch/migration.py]: no need to block on this, but it would be slightly cleaner to just return from this line

**Comment by gcapalbo** [comment]
:+1: Can we just add to this comment https://github.com/dimagi/dimagi-utils/blob/master/dimagi/utils/couch/migration.py#L22, noting now that we should make sure neither of the models override the delete() method for the same reason?

**Comment by orangejenny** [comment]
@gcapalbo good point, updated

**Pull request merged**